examples/src/example-udp-input.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * example-udp-input
+ *
+ * This example demonstrates how to use the UDP Core Input
+
+ * DEPENDENCIES:
+ * - log4js - is a generic logger for NodeJS. See the file ./providers/custom_logger.js for a basic
+ *   example of log4js as a replacement to console.log.
+ *
+ */
+
+// Bring in Package Dependencies
+const { default: RulesHarvester, CoreInputUdp } = require('rule-harvester');
+
+// Bring in other Application Specific dependencies
+const RuleOutputProviderFile = require('./providers/rule_output_file');
+const ruleClosures = require('./providers/rule_closures');
+const ruleCorpus = require('./providers/rule_corpus_udp');
+const logger = require('./providers/custom_logger');
+
+let udpFileCounter = 0;
+
+const coreUdpInput = new CoreInputUdp([3333], logger, {
+  inputContextCallback: (msg) => {
+    // In this example, this callback receives the full AMQP-Cacoon message object (of type ConsumeMessage).
+    // Our other application logic (the output provider in this case) is expecting
+    // the orderFile to be set with a "name" that will be used to output a processed order to the file system.
+    return {
+      // Construct orderFile from the incoming message fields! For this, we'll just use
+      // data coming from the AMQP Broker as an example, but it could be anything!
+      orderFile: `udp_order_file-${udpFileCounter++}`,
```

Comment:
It's not clear here what udpFileCounter is or where it comes from. I see it initialized to zero but why used here? Where else is it being changed, etc? Let's add some clarity about what's going on here.

Reply:
I reworked this section to be more like the http example and fixed the comments in the process